### ISSUE TITLE
Failed to spin up Karpenter v0.12.0 by using Helm chart

### ISSUE THREAD

**Issue by nthienan** [aws/karpenter#1997]
## Version

**Karpenter**: v0.12.0

**Kubernetes**: v1.21.12-eks-a64ea69

## Expected Behavior
Karpenter pods should be running 

## Actual Behavior
Pods are in state `CrashLoopBackOff` due to error related to initially creating leader election

## Steps to Reproduce the Problem
Deploy Karpenter by using Helm chart version `v0.12.0`

## Resource Specs and Logs
Pod log:
```
E0625 07:46:20.008930       1 leaderelection.go:329] error initially creating leader election record: configmaps is forbidden: User "system:serviceaccount:kube-system:karpenter" cannot create resource "configmaps" in API group "" in the namespace "kube-system"
E0625 07:46:22.534363       1 leaderelection.go:329] error initially creating leader election record: configmaps is forbidden: User "system:serviceaccount:kube-system:karpenter" cannot create resource "configmaps" in API group "" in the namespace "kube-system"
E0625 07:46:25.628043       1 leaderelection.go:329] error initially creating leader election record: configmaps is forbidden: User "system:serviceaccount:kube-system:karpenter" cannot create resource "configmaps" in API group "" in the namespace "kube-system"
```

Role:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  annotations:
    meta.helm.sh/release-name: karpenter
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2022-06-25T07:37:27Z"
  labels:
    app.kubernetes.io/instance: karpenter
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: karpenter
    app.kubernetes.io/version: 0.12.0
    helm.sh/chart: karpenter-0.12.0
  name: karpenter
  namespace: kube-system
  resourceVersion: "1990"
  uid: 6b8a4114-bc00-46ba-9d6b-3a8374cd667a
rules:
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - watch
- apiGroups:
  - ""
  resources:
  - configmaps
  - namespaces
  - secrets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
- apiGroups:
  - ""
  resourceNames:
  - karpenter-cert
  resources:
  - secrets
  verbs:
  - update
- apiGroups:
  - ""
  resourceNames:
  - karpenter-global-settings
  - karpenter-leader-election
  - config-logging
  resources:
  - configmaps
  verbs:
  - create
  - update
  - patch
  - delete
- apiGroups:
  - coordination.k8s.io
  resourceNames:
  - karpenter-leader-election
  - webhook.configmapwebhook.00-of-01
  - webhook.defaultingwebhook.00-of-01
  - webhook.validationwebhook.00-of-01
  - webhook.webhookcertificates.00-of-01
  resources:
  - leases
  verbs:
  - create
  - patch
  - update
```

RoleBinding:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  annotations:
    meta.helm.sh/release-name: karpenter
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2022-06-25T07:37:27Z"
  labels:
    app.kubernetes.io/instance: karpenter
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: karpenter
    app.kubernetes.io/version: 0.12.0
    helm.sh/chart: karpenter-0.12.0
  name: karpenter
  namespace: kube-system
  resourceVersion: "1993"
  uid: dbcdbcdf-bb7a-4e01-828d-40727a0a08f5
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: karpenter
subjects:
- kind: ServiceAccount
  name: karpenter
  namespace: kube-system
```

ServiceAccount:
```
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::111111111111:role/KarpenterAddonsKarpenterServiceA-1Q4YL7TO8I3GI
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{"eks.amazonaws.com/role-arn":"arn:aws:iam::111111111111:role/KarpenterAddonsKarpenterServiceA-1Q4YL7TO8I3GI"},"labels":{"app.kubernetes.io/name":"karpenter","aws.cdk.eks/prune-c83d68173c533a786e34f2ee0ce6c38b7253115e6f":""},"name":"karpenter","namespace":"kube-system"}}
  creationTimestamp: "2022-06-25T07:37:10Z"
  labels:
    app.kubernetes.io/name: karpenter
    aws.cdk.eks/prune-c83d68173c533a786e34f2ee0ce6c38b7253115e6f: ""
  name: karpenter
  namespace: kube-system
  resourceVersion: "1923"
  uid: 55387742-e618-448f-8d6f-56fe900660fa
secrets:
- name: karpenter-token-knh5j
```

**Comment by nthienan** [comment]
the issue doesn't happen on version `0.11.1`

**Comment by FernandoMiguel** [comment]
known issue https://github.com/aws/karpenter/pull/1984
fix is merged into main but no release has been cut yet

**Comment by universam1** [comment]
Same issue here, can we release the fixed chart asap please?

**Comment by universam1** [comment]
@FernandoMiguel @bwagner5 This issue is _not_ fixed by #1984 - tested in EKS cluster. 
The missing permission is:
```
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - create
```

**Comment by FernandoMiguel** [comment]
> @FernandoMiguel @bwagner5 This issue is _not_ fixed by #1984 - tested in EKS cluster. 
> The missing permission is:
> ```
> - apiGroups:
>   - ""
>   resources:
>   - configmaps
>   verbs:
>   - create
> ```

@universam1 you tested main branch?

**Comment by universam1** [comment]
@FernandoMiguel yes, but notice the different `update` permission missing:
```
failed to update lock: configmaps "karpenter-leader-election" is forbidden: User "system:serviceaccount:karpenter:karpenter" cannot update resource "configmaps" in API group "" in the namespace "karpenter"
```

**Comment by dewjam** [comment]
Hello @universam1 ,
We recently switched to [lease only elections](https://github.com/aws/karpenter/pull/1990) (it's in the main branch but not yet released).  So it's possible the role in main will not work with existing released Karpenter versions.  There was another PR which meant to fix this problem with v0.12.0:  https://github.com/aws/karpenter/pull/1988 (though I think there is a typo in this PR)

If you're using the default v0.12.0 role, you can append the below to get leadership election to work once again.  Let me know if it works for you.
```
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - create
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - create
```

Here's what the whole entire role looks like if that helps.
```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  annotations:
    meta.helm.sh/release-name: karpenter
    meta.helm.sh/release-namespace: karpenter
  labels:
    app.kubernetes.io/instance: karpenter
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: karpenter
    app.kubernetes.io/version: 0.12.0
    helm.sh/chart: karpenter-0.12.0
  name: karpenter
  namespace: karpenter
rules:
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - watch
- apiGroups:
  - ""
  resources:
  - configmaps
  - namespaces
  - secrets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
- apiGroups:
  - ""
  resourceNames:
  - karpenter-cert
  resources:
  - secrets
  verbs:
  - update
- apiGroups:
  - ""
  resourceNames:
  - karpenter-global-settings
  - karpenter-leader-election
  - config-logging
  resources:
  - configmaps
  verbs:
  - update
  - patch
  - delete
- apiGroups:
  - coordination.k8s.io
  resourceNames:
  - karpenter-leader-election
  - webhook.configmapwebhook.00-of-01
  - webhook.defaultingwebhook.00-of-01
  - webhook.validationwebhook.00-of-01
  - webhook.webhookcertificates.00-of-01
  resources:
  - leases
  verbs:
  - patch
  - update
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - create
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - create
```

**Comment by njtran** [comment]
This is now fixed. https://github.com/aws/karpenter/releases/tag/v0.12.1. If you do the same helm command with v0.12.1 you should see the changes fixed.
